kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/ExecListener.java

```diff
@@ -15,32 +15,38 @@
  */
 package io.fabric8.kubernetes.client.dsl;
 
-import okhttp3.Response;
-
+import java.io.IOException;
 
 public interface ExecListener {
+  
+  public interface Response {
+    
+    int code();
+    
+    /**
+     * May be null if not provided by the underlying implementation.
+     * @return the body as a String
+     * @throws IOException
+     */
+    String body() throws IOException;
+    
+  }
 
   /**
    * Called when the request has successfully been upgraded to a web socket.
-   *
-   * @param response OkHttp response object
    */
-  void onOpen( Response response);
-
-    /**
-     * Called when the transport or protocol layer of this web socket errors during communication.
-     *
-     * @param t Throwable
-     * @param response Present when the failure is a direct result of the response (e.g., failed
-     * upgrade, non-101 response code, etc.). {@code null} otherwise.
-     */
-    void onFailure(Throwable t, Response response);
+  default void onOpen() {}
```

Comment:
OK, was asking because I noticed one of our example operator in the operator SDK was failing to compile with this PR because it's using a listener. It uses the response's message for some logging, probably nothing critical, just don't have experience with this specific feature so was wondering the reason behind the change. Thanks!